docs/versions/index.md

```diff
@@ -0,0 +1,124 @@
+* **Deployment type:** The way Elastic is deployed: 
+  * **Serverless** (sometimes "Elastic Cloud Serverless")
+  * **Elastic Cloud Hosted**
+  * **Elastic Cloud on Kubernetes**
+  * **Elastic Cloud Enterprise**
+  * **Self-managed**
+
+  All deployment types other than **Serverless** use the **Stack <version>** flavor of Elasticsearch / Kibana.
+
+% TODO: Final term for "Self-managed"
+
+* **Project type:** The Serverless project types where a feature can be used - either **Elasticsearch**, **Security**, or **Observability**.
+
+* **Other versioning schemes:** Elastic products or tools with a versioned component, where stack versioning is not followed. 
+  
+  E.g. clients, Elastic Common Schema
+
+**How many facets do I need to use?**
+
+The role of these labels is providing a trust signal that the reader is viewing content that’s applicable to them. This means that the relevant labels should appear on all pages. However, we can choose to expose only one versioning plane on pages where only one plane is relevant:
+
+* Depending on what you're documenting, you might need to include information from multiple facets. For example, when relevant features exist at both the stack and the deployment level, both of these groups might be used together (e.g. security, user management, trust).
+
+* In some sections, such as **Explore and analyze**, features generally only differ by Elasticsearch flavor. In these cases, you can choose to include only this facet on the page.
+
+### Defaults and hierarchy 
+
+You should not assume a default deployment type or Elasticsearch / Kibana flavor.
+
+Treat all flavors and deployment types equally. Don't treat one as the "base" and the other as the "exception".
+
+However, we should all display our flavors and versions in the same order for consistency. This order reflects organizational priorities.
+
+**Flavors:**
+
+1. Serverless
+2. Stack
+
+**Deployment types:**
+
+1. Serverless
+2. Elastic Cloud Hosted
+3. Elastic Cloud on Kubernetes
+4. Elastic Cloud Enterprise
+5. Self-managed
+
+When it comes to hierarchy of versions, always put the newest version first.
+
+### Versions and lifecycle states
+
+In the V3 docs system, we currently document only our latest major versions (Stack 9.0+, ECE 4.0+, ECK 3.0+).
+
+Add version labels only when a feature was added as part of the major version release, or added in subsequent dot releases. Do not add version information to features added before these releases. For example, features added in 8.16 don't need an 8.16 label or a 9.0 label, but features added in 9.0 need a 9.0 label.
+
+From the latest major onward, the entire feature lifecycle should be represented. This means that you need to add separate labels for the following states:
+
+* beta or technical preview
+* GA
+* deprecated
+* removed
+
+We hope to simplify or consolidate these lifecycle labels in future.
+
+::::{warning}
+This feature doesn't currently work - currently, only one label will appear.
```

Comment:
> This feature doesn't currently work - currently, only one label will appear.

I think we should put a warning like this about the lack of an "on-prem / self-managed" deployment type too.